### PR TITLE
Release 23.05.2

### PR DESCRIPTION
**Pull request by rustyrussell**
YA bugfix release, please review carefully!

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,6 +32,14 @@ jobs:
...
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Rebase
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git fetch origin ${{ github.base_ref }}
+          git rebase origin/${{ github.base_ref }}
```
Comment (vincenzopalazzo) [inline .github/workflows/ci.yaml:42]: this is unrelated to this PR but this code does not work for user that are outside for the ElementsProject

If I merge a PR the Ci on master just fails with

```
Run git config user.name github-actions
  git config user.name github-actions
  git config user.email github-actions@github.com
  git fetch origin 
  git rebase origin/
  shell: /usr/bin/bash -e {0}
  env:
    RUST_PROFILE: release
    SLOW_MACHINE: 1
    RUST: 1
    COMPAT: 1
    BOLTDIR: bolts
  
fatal: invalid upstream 'origin/'
Error: Process completed with exit code 128.
```

File: doc/Makefile [hunk 2]
```diff
@@ -227,18 +227,19 @@ doc/index.rst: $(MANPAGES:=.md)
...
+	if git diff $(BASE_REF) -- doc/schemas | grep -q '^+.*{' && ! git diff $(BASE_REF) -- doc/schemas | grep -q '^+.*"added"'; then \
+		git diff $(BASE_REF) -- doc/schemas; \
 		echo 'New schema fields must have "added": "vNEXTVERSION"' >&2; exit 1; \
 	fi
+
 schema-removed-check:
-	@if ! test -z $$CI; then git fetch origin master; fi; \
-	if git diff origin/master -- doc/schemas | grep -q '^-.*{' && ! git diff origin/master -- doc/schemas | grep -q '^-.*"deprecated"'; then \
-		git diff origin/master -- doc/schemas ; \
+	if git show --format=%B -s $(BASE_REF).. | grep -q '^No-schema-diff-check'; then echo $@ suppressed; exit 0; fi; \
```
Comment (vincenzopalazzo) [inline doc/Makefile:240]: Smart!

**Review by vincenzopalazzo — COMMENTED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6357/commits/f960c9e87e51c43085eafe074a0ab9a83743ce61

As it is the PR, it is good and we fixed all the issues that we had recently, thanks to patching these, I had some comments but not blocking if you are planning to ship it and there is no blocking rust PR, please do without them!

----

In the commit https://github.com/ElementsProject/lightning/pull/6357/commits/c4a197571753e87278dc93acb5987186373a4cdc, could we also include the test? <https://github.com/ElementsProject/lightning/pull/6351>. But the integration testing may not be the correct place to test it.

Also, if there is still time due that we are releasing the Python packag. We can also include this https://github.com/ElementsProject/lightning/pull/6350 ?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (rustyrussell) [reply .github/workflows/ci.yaml:42]: Huh. There's probably a GH variable we should use here?

Comment (vincenzopalazzo) [reply .github/workflows/ci.yaml:42]: I have no idea, but from the following blog post https://www.jessesquires.com/blog/2021/10/17/github-actions-workflows-for-automatic-rebasing-and-merging/ I see that maybe there is no need to specify the `origin/`?

overall in the CI we should have a single remote

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6357/commits/6896e9bcddef1d52d51ded64d7bd9f2e882d3d62

**Comment by endothermicdev** [comment]
I compiled and tested locally, which was fine except for a memory leak that was reported from lightningd/onchain_control.c stemming from new_channel_subd.  Is this is a fluke of the memleak detection, or something we should address?  Doesn't seem to be related to any of these commits at least.

E           Global errors:
E            - Node /tmp/ltests-l87atiyq/test_onchain_middleman_simple_1/lightning-1/ has memory leaks: [
E               {
E                   "backtrace": [
E                       "ccan/ccan/tal/tal.c:477 (tal_alloc_)",
E                       "ccan/ccan/io/io.c:91 (io_new_conn_)",
E                       "lightningd/subd.c:774 (new_subd)",
E                       "lightningd/subd.c:828 (new_channel_subd_)",
E                       "lightningd/onchain_control.c:1418 (onchaind_funding_spent)",
E                       "lightningd/peer_control.c:1890 (funding_spent)",
E                       "lightningd/watch.c:271 (txowatch_fire)",
E                       "lightningd/chaintopology.c:82 (filter_block_txs)",
E                       "lightningd/chaintopology.c:938 (add_tip)",
E                       "lightningd/chaintopology.c:1029 (get_new_block)",
E                       "lightningd/bitcoind.c:494 (getrawblockbyheight_callback)",
E                       "lightningd/plugin.c:590 (plugin_response_handle)",
E                       "lightningd/plugin.c:701 (plugin_read_json_one)",
E                       "lightningd/plugin.c:746 (plugin_read_json)",
E                       "ccan/ccan/io/io.c:59 (next_plan)",
E                       "ccan/ccan/io/io.c:407 (do_plan)",
E                       "ccan/ccan/io/io.c:417 (io_ready)",
E                       "ccan/ccan/io/poll.c:453 (io_loop)",
E                       "lightningd/io_loop_with_timers.c:22 (io_loop_with_timers)",
E                       "lightningd/lightningd.c:1231 (main)",
E                       "../sysdeps/nptl/libc_start_call_main.h:58 (__libc_start_call_main)",
E                       "../csu/libc-start.c:392 (__libc_start_main_impl)"
E                   ],
E                   "label": "ccan/ccan/io/io.c:91:struct io_conn",
E                   "parents": [
E                       "lightningd/lightningd.c:106:struct lightningd"
E                   ],
E                   "value": "0x55ebc0225e78"
E               }
E           ]

Looks good to me otherwise.

**Comment by rustyrussell** [comment]
Yes, this is a temporary false positive, all good.

**Pull request merged**